purchase_allowed_product/tests/test_purchase_allowed_product.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2017 Today Mourad EL HADJ MIMOUNE @ Akretion
+# Copyright 2020 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo.tests import Form
+
+from odoo.addons.base.tests.common import BaseCommon
+
+
+class TestPurchaseAllowedProduct(BaseCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.supplierinfo_model = cls.env["product.supplierinfo"]
+        cls.product_model = cls.env["product.product"]
+        cls.partner_4 = cls.env.ref("base.res_partner_4")
```

Comment:
```suggestion
        cls.partner = cls.env["res.partner"].create({"name": "Test supplier"})
```
Let's try to avoid using demo data to prevent possible side effects.

Reply:
Thanks @victoralmau for the suggestion. I updated the test to create the partner directly instead of using demo data, as recommended.